### PR TITLE
Always program OpenStack floating IPs

### PR DESCRIPTION
**Pull request by neiljerram**
## Description

Felix's programming of NAT mappings was disabled by default in https://github.com/projectcalico/calico/pull/5861.  This was a security fix for Kubernetes, as in Kubernetes anyone could create a floating IP for themselves by adding a "floatingips" annotation to their own pods.

For OpenStack, pod annotations obviously do not apply, and NAT mappings come from the ML2 driver running as part of the Neutron server, which is already access-protected in an appropriate way.  And we don't want to have to document that OpenStack users need to add "FloatingIPs = Enabled" to their felix.cfg file.

Therefore, arrange for OpenStack-configured floating IPs to be programmed regardless of the FloatingIPs setting in FelixConfiguration.

## Related issues/PRs

qualifies https://github.com/projectcalico/calico/pull/5861

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/dataplane/linux/endpoint_mgr.go [hunk 1]
```diff
@@ -658,13 +659,9 @@ func (m *endpointManager) resolveWorkloadEndpoints() {
...
-				if m.floatingIPsEnabled && len(natInfos) != 0 {
-					// Include any floating IP NATs if the feature is enabled.
-					old := ipStrings
-					ipStrings = make([]string, len(old)+len(natInfos))
-					copy(ipStrings, old)
-					for ii, natInfo := range natInfos {
-						ipStrings[len(old)+ii] = natInfo.ExtIp + addrSuffix
+				for _, natInfo := range natInfos {
+					if m.floatingIPsEnabled || natInfo.Source == apiv3.IPNATSourceOpenStack {
+						ipStrings = append(ipStrings, natInfo.ExtIp+addrSuffix)
```
Comment (fasaxc) [inline felix/dataplane/linux/endpoint_mgr.go:664]: Not sure if this is OK, appending to a slice can share the same backing array so if two bits of code decide to append different things to the same slice then they might see each other's updates.  Better to take your own copy?

**Review by fasaxc — CHANGES_REQUESTED** [review]
How about using the orchestrator ID in the WEP key then we don't need a datamodel change?  We've carried that ID around for years maybe it should pay its way!

**Comment by neiljerram** [comment]
Great idea about the orchestrator ID, and that should also make the change a lot smaller - I'll take a look at that.

File: felix/dataplane/linux/endpoint_mgr.go (hunk 1 above)
Comment (neiljerram) [reply felix/dataplane/linux/endpoint_mgr.go:669]: Yes, I've restored logic like we had here before.

**Review by fasaxc — APPROVED** [review]

**Pull request merged**